### PR TITLE
Gradle build improvements

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -44,7 +44,15 @@ semver {
...
-version = semver.semVersion
+// Only override a default version (which usually is "unspecified"), but not a custom version.
+if (version == Project.DEFAULT_VERSION) {
+    version = semver.semVersion.takeIf { it.isPreRelease }
+        // To get rid of a build part's "+" prefix because Docker tags do not support it, use only the original "build"
+        // part as the "pre-release" part.
+        ?.toString()?.replace("${semver.defaultPreRelease}+", "")
+            // Fall back to a plain version without pre-release or build parts.
+        ?: semver.version
+}
```
Comment (mnonnenmacher) [inline build.gradle.kts:55]: I find it difficult to understand the impact of this change. Would it affect the Docker tags or the versions of the Maven packages? What exactly changes compared to before?

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]

File: build.gradle.kts (hunk 1 above)
Comment (mnonnenmacher) [reply build.gradle.kts:52]: Thanks, but it's still not totally clear to me what the difference between before and after is for a pre release version. I think what would help me would be an example.
Actually, the reason I did not copy this code from ORT was that I was not sure if doing the fix here for Docker could negatively impact the versions of any other release artifacts.

If this is changed, probably this line from the Docker build workflow needs to be adapted:
https://github.com/eclipse-apoapsis/ort-server/blob/a72980ee6d6294a27edc9345991a216e68f9ce78/.github/workflows/docker-build.yml#L128

Commit message: Typo in "chnage"

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]

File: build.gradle.kts (hunk 1 above)
Comment (sschuberth) [reply build.gradle.kts:52]: > If this is changed, probably this line from the Docker build workflow needs to be adapted:

I believe it does not *need* to be adjusted, but the final ` | sed 's/+/-/` could be dropped. Then the code would be equals to ORT's.

> Commit message: Typo in "chnage"

Fixed.

**Review by mnonnenmacher — APPROVED** [review]

File: build.gradle.kts (hunk 1 above)
Comment (sschuberth) [reply build.gradle.kts:52]: > I think what would help me would be an example.

The code changes a pre-release version that would usually read like `26.0.0-SNAPSHOT+010.sha.72d9291` to read `26.0.0-010.sha.72d9291` instead.

**Pull request merged**